src/devices/maverick_et73x.c

```diff
@@ -0,0 +1,253 @@
+    
+    return currentValue;
+}
+
+static uint16_t calculate_checksum(uint32_t data) {
+    //initial value of linear feedback shift register
+    uint16_t mask = 0x3331; 
+    uint16_t csum = 0x0;
+    int i;
+    for(i = 0; i < 24; ++i) {
+        //data bit at current position is "1"
+        //do XOR with mask
+        if((data >> i) & 0x01)
+            csum ^= mask;
+        
+        mask = shiftreg(mask);
+    }
+    return csum;
+}
+
+static int maverick_et73x_callback(bitbuffer_t *bitbuffer) {
+    data_t *data;
+    char time_str[LOCAL_TIME_BUFLEN];
+    double diff_t = 0.0;
+    int8_t b_use_message = 0;
+    int32_t chk_xor;
+    char* dev_state;
+
+    //we need an inverted bitbuffer
+    bitbuffer_invert(bitbuffer);
+
+    if(bitbuffer->num_rows != 1)
+        return 0;
+
+    //check correct data length
+    if(bitbuffer->bits_per_row[0] != MAV_MESSAGE_LENGTH)
+        return 0;
+
+    //check for correct header (0xAA9995)
+    if((bitbuffer->bb[0][0] != 0xAA || bitbuffer->bb[0][0] != 0xaa ) || bitbuffer->bb[0][1] != 0x99 || bitbuffer->bb[0][2] != 0x95)
+        return 0;
+    
+    //convert hex values into quardinary values
+    convert_bitbuffer(bitbuffer);
+    
+    //checksum is used to represent a session. This means, we get a new session_id if a reset or battery exchange is done. 
+    chk_xor = (calculate_checksum(checksum_data()) & 0xffff) ^ checksum_received();
+    
+    
+    dev_state = get_status();
+    
+    //if the transmitter is in init state, we take the session_id for further checks. 
+    if(strncmp(dev_state, "init", 4) == 0)
+    {
+        //assuming that having two times the same checksum, means there was no error in sending first message
+        if(session_id != chk_xor)
+            b_use_message = 1;
+    }
+
+    //second message
+    if(session_id > 0) {
+        time(&time_now_t);
+        diff_t = difftime(time_now_t, last_time_t);
+
+        //checksum error
+        if(session_id != chk_xor && b_use_message == 0 && diff_t > 8)
+            return 0;
+    }
+    
+
+    //one message (of four) is enough for output
+    if (diff_t < 8 && session_id > 0 && b_use_message == 0)
+        return 0;
+    
+    session_id = chk_xor;
+    time(&last_time_t);
+
+    local_time_str(0, time_str);
+
+    if(debug_output)
+        fprintf(stderr, "checksum xor: %x\n", chk_xor);
+
+    data = data_make("time",           "",                      DATA_STRING,                         time_str,
+                     "id",             "Session_ID",            DATA_INT,                            chk_xor,
+                     "temperature_C1", "TemperatureSensor1",    DATA_FORMAT, "%.02f C", DATA_DOUBLE, get_temperature(TEMPERATURE_START_POSITION_S1),
+                     "temperature_C2", "TemperatureSensor2",    DATA_FORMAT, "%.02f C", DATA_DOUBLE, get_temperature(TEMPERATURE_START_POSITION_S2),
+                     "status",         "Status",                DATA_STRING,                         dev_state,
```

Comment:
Here you should add a crc member to inform that the message was received with crc.